mycroft/api/__init__.py

```diff
@@ -497,19 +497,33 @@ def is_paired(ignore_errors=True):
         # The Mark 1 does perform a restart on RESET.
         return True
 
+    api = DeviceApi()
+    _paired_cache = api.identity.uuid and check_remote_pairing(ignore_errors)
+
+    return _paired_cache
+
+
+def check_remote_pairing(ignore_errors):
+    """ Check that a basic backend endpoint accepts our pairing.
+
+    Arguments:
+        ignore_errors (bool): True if errors should be ignored when
+
+    Returns: True if pairing is checks out, otherwise False.
+    """
     try:
         api = DeviceApi()
         device = api.get()
-        _paired_cache = api.identity.uuid is not None and \
-            api.identity.uuid != ""
-        return _paired_cache
     except HTTPError as e:
         if e.response.status_code == 401:
             return False
     except Exception as e:
         LOG.warning('Could not get device info: ' + repr(e))
-    if ignore_errors:
-        return False
-    if connected():
-        raise BackendDown
-    raise InternetDown
+        if ignore_errors:
```

Comment:
I am not sure that moving these lines inside the exception block is a good idea.  if the backend throws a 404 or 500 error, that would trigger the HTTPError exception and these lines of code would not be executed.

Reply:
Good catch

Reply:
I've updated the section a bit now. Not sure if it makes things worse or better :)